pkg/apis/admissionregistration/types.go

```diff
@@ -309,6 +309,11 @@ type WebhookClientConfig struct {
 	// the webhook's server certificate.
 	// Required.
 	CABundle []byte
+
+	// `timeout` is the number of seconds after which the webhook request times out.
+	// Must be greater than 0 if set. If not set, will use server default 30s.
+	// +optional
+	Timeout *metav1.Duration
```

Comment:
IMO we should not add this to the API. Why do different webhooks need different timeouts? Webhooks should all return in milliseconds, always. My gut instinct is that there should be a global 1s timeout.

Reply:
In case webhook server is under great pressure or we may need more complicated logic in webhook, which all lead to not response in 1s.

Make it configurable is more flexible for users.

Reply:
Not all flexibility is equally useful, but it all comes with a maintenance cost. Can you give a more concrete use case? My gut reaction to both items you list is that the webhook authors need to fix their webhook.